src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -547,41 +545,130 @@ private void SetDirty(TimeSpan? maxDelay)
             return applicationData;
         }
 
-        private async Task InitializeAsync(bool ignoreFailures, IEnumerable<ConfigurationClient> availableClients, CancellationToken cancellationToken = default)
+        private async Task LoadAsync(bool ignoreFailures, CancellationToken cancellationToken)
         {
-            Dictionary<string, ConfigurationSetting> data = null;
-            Dictionary<KeyValueIdentifier, ConfigurationSetting> watchedSettings = null;
-            
+            var startupStopwatch = Stopwatch.StartNew();
+
+            int exponentialBackoffAttempts = 0;
+
+            var startupExceptions = new List<Exception>();
+
+            IEnumerable<ConfigurationClient> clients = _configClientManager.GetAllClients();
+
             try
             {
-                await ExecuteWithFailOverPolicyAsync(
-                    availableClients,
-                    async (client) =>
+                while (true)
+                {
+                    if (await TryInitializeAsync(clients, startupExceptions, cancellationToken).ConfigureAwait(false))
+                    {
+                        break;
+                    }
+
+                    try
+                    {
+                        if (startupStopwatch.Elapsed < FailOverConstants.StartupFixedBackoffDuration)
+                        {
+                            await Task.Delay(startupStopwatch.Elapsed.CalculateFixedStartupBackoffDuration(), cancellationToken).ConfigureAwait(false);
```

Comment:
the timespan you are passing to the method has a different value than the one you have checked in the condition.

Reply:
To avoid the problem, you could cache elapsed in a variable, but I'd recommend an approach like

```
TimeSpan delay;
if (startupStopwatch.TryGetFixedBackoff(out Timespan backoff))
{
    delay = backoff;
}
else
{
    postFixedWindowAttempts++;

    delay = FailoverConstants.MaxFixedStartupBackoff.CalculateBackoff(
        attempts: postFixedWindowAttempts,
        maxDuration: FailOverConstants.MaxBackoffDuration
    );
}

await Task.Delay(delay, ct);

```